src/Shared/Domain/ValueObject/DateTimeValueObject.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace CodelyTv\Shared\Domain\ValueObject;
+
+abstract class DateTimeValueObject
```

Comment:
This is not needed, you could treat `DateTimeImmutable` as a scalar, like `string`